lib/index.js

```diff
@@ -13,16 +13,16 @@ if (require.main === module) {
 
     load (configPath) {
       this.configPath = configPath;
+      process.env.EG_CONFIG_DIR =
+      this.configPath || process.env.EG_CONFIG_DIR;
```

Comment:
Yea I see what you're saying.. config not matching what's happening could be pretty confusing.

This does look like it would work functionally though.

What if we stole a page from Jenkins.. in a Jenkinsfile, if you want to reference something defined in Jenkins Global Environment, it's referenced under the `env` key.

So in your Jenkinsfile you have Strings like "${env.monitorDomain}".

If we provided the config with a `secrets` object, then from the YAML file, those secrets could be referenced, and an error can be thrown if one is referenced but not defined, or fallback's could be provided like env the go syntax/docker-compose var's `${ENV_NAME:-default}`.

That way if you inspected the file, it would match what is running.

```
# Core
db:
  redis:
    host: ${REDIS_HOST:-redis}
    port: ${REDIS_PORT:-6379}
    namespace: EG

crypto:
  cipherKey: sensitiveKey
  algorithm: aes256
  saltRounds: 10

plugins:
  rewrite:
    package: express-gateway-plugin-rewrite

session: 
  storeProvider: connect-redis
  storeOptions:
    host: localhost
    port: 6379
  secret: ${secrets.session:-adefaultsecret} # replace with secure key that will be used to sign session cookie
  resave: false
  saveUninitialized: false
```